plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/ProjectStatePersister.java

```diff
@@ -125,21 +126,28 @@ public ProjectState copy() {
 	/** The current version of the persistence format. Increment to support backwards compatible deserialization. */
 	private static final int CURRENT_VERSION = VERSION_2;
 
-	/** The simple name of the file with the project state. */
-	public static final String FILENAME = N4JSGlobals.N4JS_PROJECT_STATE;
-
 	private final ExecutorService writer = Executors.newSingleThreadExecutor();
 
+	/** @return the simple name of the file with the project state. */
+	public String getPersistedFileName() {
+		return ".projectstate";
+	}
+
 	/**
-	 * Close this persister and wait for pending write operations to complete.
+	 * The language version changes iff any of the persisted languages changes the serialization of its persisted state.
```

Comment:
Isn't it rather: "The language version must change iff ..." or "... must be changed iff ..."? Or "A different language version must be passed in iff ..."?

Reply:
IMHO, that is all the same. My version is more declarative, yours is more imperative.